tests/typed_search.test.ts

```diff
@@ -352,11 +348,28 @@ describe.each([
   test(`${permission} key: Search on index with no documents and no primary key`, async () => {
     const client = await getClient(permission)
     const response = await client.index(emptyIndex.uid).search('prince', {})
-    expect(response.limit === 20).toBeTruthy()
+
+    expect(response.hits.length === 0).toBeTruthy()
     expect(response).toHaveProperty('processingTimeMs', expect.any(Number))
     expect(response.query === 'prince').toBeTruthy()
   })
 
+  test(`${permission} key: search with pagination parameters hitsPerPage/page and offset`, async () => {
+    const client = await getClient(permission)
+
+    const response = await client.index<Movie>(index.uid).search('', {
+      hitsPerPage: 1,
+      page: 1,
+      limit: 1,
+    })
+
+    expect(response.hits.length).toEqual(1)
+    expect(response.hitsPerPage === 1).toBeTruthy()
+    expect(response.page === 1).toBeTruthy()
+    expect(response.totalPages === 7).toBeTruthy()
+    expect(response.totalHits === 7).toBeTruthy()
```

Comment:
It should only respond with the following:

```json
{
    "hits": [
        {
           ...
        }
    ],
    "query": "",
    "processingTimeMs": 0,
    "limit": 1,
    "offset": 0,
    "estimatedTotalHits": xx
}
```

So your test should match the estimated pagination response.